workflow/controller/operator.go

```diff
@@ -82,15 +81,21 @@ func (wfc *WorkflowController) operateWorkflow(wf *wfv1.Workflow) {
 			woc.markWorkflowFailed(fmt.Sprintf("invalid spec: %s", err.Error()))
 			return
 		}
+	} else {
+		err = woc.podReconciliation()
+		if err != nil {
+			woc.log.Errorf("%s error: %+v", wf.ObjectMeta.Name, err)
+			// TODO: we need to re-add to the workqueue, but should happen in caller
+			return
+		}
 	}
 
-	err := woc.createPVCs()
+	err = woc.createPVCs()
```

Comment:
In a separate commit, should this code be refactored? Shouldn't the call to createPVCs only when the workflow starts?

Reply:
If we get here, we are just about to start scheduling pods as part of the workflow. Where would this be moved?

Reply:
Isn't operateWorkflow() called for every update?

runWorker -> processNextItem -> operateWorkflow?

Reply:
Yes, but createPVCs is idempotent and becomes a no-op on subsequent operateWorkflow() calls.

Reply:
Right. It just makes it harder to read and understand.